src/lib/common/sol-interrupt_scheduler_riot.c

```diff
@@ -40,8 +40,10 @@ static kernel_pid_t pid;
 
 enum interrupt_type {
     GPIO,
```

Comment:
if gpio is ripped out the gpio scheduler will never get called ever - well uart will never be called as well - but in the case of uart it depends on uart_t which will not be present in that case (if the platform doesn't support or implement it) that's the why I'm ifdef'ing this code here, actually this enum could not even be ifdef'ed but for "consistency sake" I'm also guarding it... if you think we should keep consistency between gpio and uart we could ifdef this GPIO enum as well, but it's not required.